src/hooks.js

```diff
@@ -61,17 +58,17 @@ export const usePreloadDogs = () => {
       preloadUrls(dogUrls)
 
       setPreloadDogs((preloadDogs) => {
```

Comment:
Should we write setPreloadDogs as below?

setPreloadDogs(() => {
        return [...dogUrls, ...preloadDogs]
      })

Reply:
@wood-crafter 
Yeah, generally we try to avoid all direct mutations in React...!

Reply:
@wood-crafter 
Why you say this: 'generally we try to avoid all direct mutations in React'? Any related to my suggestion?

Reply:
@wood-crafter recall state APIs in React vs Vue:  

React's `setState` accept a new object instead of directly mutating the state:
```js
setItems(items => [...items, newItem])
```
Vue.js mutate the state directly (because internally it uses JS's proxy API):
```js
this.items.push(newItem)
```

Reply:
@wood-crafter a program that never changes its state directly is easier to reason about, because of fewer moving parts.

Reply:
@wood-crafter U should probably read this...!
https://reactjs.org/docs/optimizing-performance.html#the-power-of-not-mutating-data

Reply:
@wood-crafter  
> Any related to my suggestion?

These Array methods **mutate** the original array:
```js
[].concat()
[].push()
[].shift()
[].unshift()
[].pop()
// And others
```

> Avoid these, return a new array instead (with spread syntax, `.filter()`, `.map()`,...)